agent/php_stacked_segment.h

```diff
@@ -155,6 +155,397 @@
+ * know no segment caught the exception. We'll apply the acception and
+ * keep/close stacked segments all the way down the stack to clean up dangling
+ * segments. We pop the current segment *C and apply the exception. Because it
+ * has an exception, the segment is kept so we copy the contents of the stacked
+ * segment *C into a segment c we obtained from the slab allocator, and we make
+ * c a child of the stacked segment *B which becomes the current segment.
+ *
+ *  root                        root <                        root
+ *   |                           |                             |
+ *   *A <                        a                             a
+ *    |                          |                             |
+ *    b                          b                             b
+ *    |                          |                             |
+ *    c                          c                             c
+ *
+ * We pop the current segment *B and apply the exception. Because it has an
+ * exception, the segment is kept so we copy the contents of the stacked segment
+ * *B into a segment b we obtained from the slab allocator, and we make b a
+ * child of the stacked segment *A which becomes the current segment.  Then we
+ * pop the current segment *A and apply the exception. Because it has an
+ * exception, the segment is kept so we copy the contents of the stacked segment
+ * *A into a segment a we obtained from the slab allocator, and we make a a
+ * child of the root.  The exception is applied to the root and the txn ends.
+ *
+ * case 4 php_observer_handle_exception_hook - when a new exception is noticed
+ *  root <                      root                      root
+ *                               |                         |
+ *                               *A <                      *A
+ *                                                         |
+ *                                                         *B
+ *
+ * We start out with a root segment, OAPI calls nr_php_observer_fcall_begin(A),
+ * and it starts stacked segment *A and then nr_php_observer_fcall_begin(B)
+ * starts stacked segment *B as a child of *A.
+ *
+ *  root                        root                       root
+ *   |                           |                           |
+ *   *A <                        *A                          *A
+ *                               |                           |
+ *                               *B                          *B <
+ *                               |                           |
+ *                               *C <                        c
+ *
+ * nr_php_observer_fcall_begin(C) starts *C gets started as child
+ * of *B. Function C throws an uncaught exception which B does not catch so
+ * netiher nr_php_observer_fcall_end(B) nor nr_php_observer_fcall_end(C) is
+ * called and *C remains the current segment. B catches the exception throws
+ * another exception. At this point we realize the current
```

Comment:
For the line 
```c
B catches the exception throws another exception.
```
Is the exception `B` catches here a separate exception to the one thrown by `C`, since `B` did not catch `C`?

Reply:
Please see: [2cb814c](https://github.com/newrelic/newrelic-php-agent/pull/580/commits/2cb814c660a71927b9287e3474bb7425bd2f45dc)